src/System.Net.Http/src/System/Net/Http/HttpContent.cs

```diff
@@ -315,8 +310,11 @@ public Task CopyToAsync(Stream stream, TransportContext context)
                 Task task = null;
                 if (IsBuffered)
                 {
-                    byte[] data = this.GetDataBuffer(_bufferedContent);
-                    task = stream.WriteAsync(data, 0, (int)_bufferedContent.Length);
+                    task = _bufferedContent.CopyToAsync(stream).ContinueWithStandard(copyTask =>
+                    {
+                        // Reset the position of _bufferedContent
+                        _bufferedContent.Position = 0;
```

Comment:
Should `_bufferedContent` be passed as state to allow the delegate to be created once and cached, instead of being allocated each time?